lib/setup.js

```diff
@@ -22,17 +22,24 @@ var constructResponse = function (server) {
   var connections = server.connections.map(function (connection) {
     var plugins = {};
 
-    // TODO: Don't use _registrations and instead use a public API
-    // related issue: https://github.com/hapijs/hapi/issues/2777
-    Object.keys(connection._registrations).forEach(function (name) {
-      var registration = connection._registrations[name];
+    Object.keys(connection.registrations).forEach(function (name) {
+      var registration = connection.registrations[name];
+
+      /* This is to step around a code choice inside hapi. There is some code in
+       * plugin.js that does a Joi validation and gives registration.attributes.pkg.version
+       * a defualt value of 0.0.0 if not set, ignoreing the one directly on the registration
+       * function */
+
+      var version = registration.version;
+      var pkg = Hoek.clone(registration.attributes.pkg);
```

Comment:
Can you add a TODO comment saying that, so it doesn't get lost. I'd like to do the ES6 jump in the next major (aka once the UI is done).